### PR TITLE
Fix kosmos

### PR DESCRIPTION
**Pull request by Cyrilvallez**
[CPU CI](https://transformers-ci.lor-e.huggingface.cool/d/pytest-observability-by-pr/pytest-observability-branch?var-pr=48853) [GPU run-slow](https://transformers-ci.lor-e.huggingface.cool/d/pytest-observability-by-pr/pytest-observability-branch?var-pr=48853)

# What does this PR do?

As per the title. Small fix after https://github.com/huggingface/transformers/pull/48814 because the mask can now be None sometimes if no padding

### CODE REVIEW

**Comment by ydshieh** [comment]
@askserge review

**Comment by ydshieh** [comment]
run-slow: kosmos2

**Comment by ydshieh** [comment]
Would still love  to have a short comment, even just like " ... because `attention_mask` is dropped earlier ..." etc, or like a failing tset with error.

A bit of context is always good, and you can even let the agent to write that short comment for you 🙏

**Comment by Cyrilvallez** [comment]
>Would still love to have a short comment

IMO it really does not warrant one, there are many many ways of getting those values, input_ids being the most obvious one IMO, and saying "we don't use mask because X" is super weird for future people reading the code, as the code actually does not touch the mask haha

**Review by ydshieh — APPROVED** [review]
